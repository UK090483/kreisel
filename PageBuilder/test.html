<style>
    .crArchive {
        border: 1px solid #888;
        background-color: #eeeeee;
    }

    .crArchiveItem {
        background-color: #ffffff;
        padding: 2px;
        text-align: left;
    }

    .crArchiveItem.CREven {
        background-color: #eeeeee;
    }

    .crArchiveItemDate {
        display: block;
        float: left;
        min-width: 85px;
        text-align: left;
    }

    .crArchiveItemName {
        display: inline;
        font-weight: bold;
    }

    .crArchiveItemImage,
    .crArchiveItemDescription {
        display: none;
    }
</style>
<script
    type='text/javascript'>var CRCategories = ''; var CRLinkTarget = '_BLANK'; var CRMaxItems = 100; var CRShowImage = true; var CRDataURL = 'https://s3-eu-west-1.amazonaws.com/files.crsend.com/114000/114707/rss/mailings.js';</script>
<script type='text/javascript' src='//s3-eu-west-1.amazonaws.com/cloud-files.crsend.com/js/rss_archive.js'></script>