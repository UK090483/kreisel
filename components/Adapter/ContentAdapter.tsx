import Content from "components/Atoms/Content";
import RichText, {
  PortableTextSource,
} from "PageBuilder/RichText/PortableText";

import React from "react";

const ContentAdapter = (props: React.ComponentProps<typeof Content>) => {
  return <RichText {...props} />;
};

export const validateContentSource = (content: any) => {
  const _content = content as PortableTextSource;
  return !!_content && Array.isArray(_content) && _content.length > 0;
};

export default ContentAdapter;
