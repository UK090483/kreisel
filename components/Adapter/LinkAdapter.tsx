import { LinkProps } from "components/Atoms/Link";
import { LinkResult } from "PageBuilder/Navigation/navigation.query";
import React, { PropsWithChildren } from "react";
import NextLink from "next/link";

export type LinkSource = LinkResult;

const LinkAdapter: React.FC<PropsWithChildren<LinkProps>> = ({
  children,
  className,
  onClick,
  href,
  external,
}) => {
  if (external) {
    return (
      <a href={href} className={className} target="_blank" rel="noreferrer">
        {children}
      </a>
    );
  }

  return (
    <NextLink
      onClick={onClick}
      data-testid="nextLink"
      href={href || "/"}
      passHref
      className={className}
    >
      {children}
    </NextLink>
  );
};

export default LinkAdapter;
