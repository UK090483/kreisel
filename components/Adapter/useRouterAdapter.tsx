import { useRouter as useNextRouter } from "next/router";

function useRouter() {
  const router = useNextRouter();
  return router;
}

export default useRouter;
