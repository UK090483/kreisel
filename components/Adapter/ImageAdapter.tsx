import Image from "../Atoms/Image";
import SanityImage from "PageBuilder/Image/frontend/SanityImage";
import { ImageResult } from "PageBuilder/baseQueries";
import React from "react";

export type imageSource = ImageResult | undefined | null;

const ImageAdapter = (props: React.ComponentProps<typeof Image>) => {
  // @ts-ignore
  return <SanityImage {...props} />;
};
export const validateSrc = (src: unknown) => {
  const _src = src as ImageResult;
  return Boolean(_src?.url);
};

export default ImageAdapter;
