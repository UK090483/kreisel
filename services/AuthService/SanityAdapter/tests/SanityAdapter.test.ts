import Adapter from "../SanityAdapter";
import { mockClient } from "../../../SanityService/test/testClient";
import {omit} from 'lodash'
const testUser1 = {
  _type: "user",
  emailVerified: new Date(2018, 1, 12, 10, 30),
  _id: "testUser1",
  name: "testName",
  email: "web@konradullrich.com",
};
const testAccount = {
  _type: "account",
  provider: "testProvider",
  providerAccountId: "testProviderAccountId",
  _id: "testAccount_1",
  user: { _ref: "testUser1" },
};

const database: any[] = [testUser1, testAccount];

const testAdapter = (db?:any[]) => {
  return Adapter({ client: mockClient({ database:db ||database }) });
};

describe("Sanity Adapter", () => {
  let client = mockClient({database})
   let adapter = Adapter({client}) 
  beforeEach(()=>{
    client = mockClient({database})
    adapter = Adapter({client}) 
  })
  it("createUser should create and return user ", async () => {
  
    const newUser = await adapter.createUser(omit(testUser1,'_id'))
    expect(newUser).toHaveProperty('id')
    expect(client.getDb().length).toBe(3);
    
  });

  //getUser ------------------------------------------------

  it("getUser should return user if exist", async () => {
    const user = await adapter.getUser("testUser1");
    expect(user).toMatchSnapshot();
  });
  it("getUser should return null if not exist", async () => {
    const user = await adapter.getUser("testUser-null");
    expect(user).toStrictEqual(null);
  });

  //getUserByEmail ------------------------------------------------

  it("getUserByEmail should return user if exist", async () => {
    const user = await adapter.getUserByEmail("web@konradullrich.com");
    expect(user).toMatchSnapshot();
  });
  it("getUserByEmail should return null if not exist", async () => {
    const user = await adapter.getUserByEmail("test@konradullrich.com");
    expect(user).toStrictEqual(null);
  });

  //getUserByAccount  ------------------------------------------------
  it("getUserByEmail should return user if exist", async () => {
    const user = await adapter.getUserByAccount({
      provider: "testProvider",
      providerAccountId: "testProviderAccountId",
    });
    expect(user).toMatchSnapshot();
  });


  // updateUser ------------------------------------------------
  it("getUserByEmail should return user if exist", async () => {
    const updated = { ...omit(testUser1,'_id'),id:'testUser1', name: "updatedName" };
    const updatedUser = await adapter.updateUser(updated);
    expect(updatedUser).toStrictEqual(updated);
    expect(client.getDb().find(i=>i._id === 'testUser1')).toStrictEqual({...testUser1,name: "updatedName"});
  });

 
});

export {};
