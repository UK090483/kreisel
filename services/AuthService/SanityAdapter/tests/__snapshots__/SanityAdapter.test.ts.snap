// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanity Adapter getUser should return user if exist 1`] = `
Object {
  "_type": "user",
  "email": "web@konradullrich.com",
  "emailVerified": 2018-02-12T09:30:00.000Z,
  "id": "testUser1",
  "name": "testName",
}
`;

exports[`Sanity Adapter getUserByEmail should return user if exist 1`] = `
Object {
  "_type": "user",
  "email": "web@konradullrich.com",
  "emailVerified": 2018-02-12T09:30:00.000Z,
  "id": "testUser1",
  "name": "testName",
}
`;

exports[`Sanity Adapter getUserByEmail should return user if exist 2`] = `
Object {
  "_type": "user",
  "email": "web@konradullrich.com",
  "emailVerified": 2018-02-12T09:30:00.000Z,
  "id": "testUser1",
  "name": "testName",
}
`;
